packages/ember-data/tests/unit/states-test.js

```diff
@@ -87,3 +87,14 @@ test("the deleted.saved state", function() {
   isFalse("isSaving");
   isTrue("isDeleted");
 });
+
+module("unit/states - hooks for record states", {
+  setup: function() {
+    rootState = DS.RootState;
+  }
+});
+
+test("root.deleted.uncommitted supports pushedData", function() {
+  equal(typeof get(rootState, "deleted.uncommitted.pushedData"), 'function',
+    "root.deleted.uncommitted should be a function");
+});
```

Comment:
Instead of testing that the `deleted.uncommitted.pushedData` function exists could you add a new test to the `model-test.js` file that creates a record with an id using `store.push`, calls `record.deleteRecord()` to move it into the `deleted.uncommitted` state then call `store.push` again with the same id then assert that the record is still in the `root.deleted.uncommitted` state and no errors are thrown?

This should help make the test more resilient against future refactors.